src/js/angular/ttyg/directives/chat-item-detail.directive.js

```diff
@@ -50,13 +57,78 @@ function ChatItemDetailComponent(toastr, $translate, TTYGContextService, Markdow
                 $scope.onRegenerateQuestion({chatItem: $scope.chatItemDetail});
             };
 
-            $scope.copyAnswerToClipboard = () => {
-                $scope.onCopyAnswerToClipboard({chatItem: $scope.chatItemDetail});
+            /**
+             * Copies the message to the clipboard.
+             *
+             * @param {string} message
+             */
+            $scope.copy = (message) => {
+                TTYGContextService.emit(TTYGEventName.COPY_TEXT_TO_CLIPBOARD, message);
+            };
+
+            /**
+             * Extract the explanation of how the answer was generated.
+             */
+            $scope.explainResponse = () => {
+                const explainResponse = TTYGContextService.getExplainResponseCache()[$scope.chatItemDetail.answer.id];
+                if (explainResponse) {
+                    explainResponse.expanded = !explainResponse.expanded;
+                    TTYGContextService.updateExplainResponseCache(explainResponse);
+                } else {
+                    $scope.loadingExplainResponse = true;
+                    TTYGService.explainResponse($scope.chatItemDetail)
+                        .then((explainResponse) => {
+                            $scope.explainResponseModel = explainResponse;
+                            TTYGContextService.updateExplainResponseCache(explainResponse);
+                        })
+                        .catch((error) => {
+                            toastr.error(getError(error.data), $translate.instant('similarity.get.resource.error'));
+                        })
+                        .finally(() => $scope.loadingExplainResponse = false);
+                }
+            };
+
+            /**
+             * Triggers an asking how the answer was generated.
+             */
+            $scope.askHowDerivedAnswer = () => {
+                $scope.onAskHowDeliveredAnswer({chatItem: $scope.chatItemDetail});
+            };
+
+            /**
+             * Opens <code>query</code> in sparql editor.
+             * @param {string} query
+             */
+            $scope.openInSparqlEditor = (query) => {
+                if ($scope.chatItemDetail.answer) {
+                    const agents = TTYGContextService.getAgents();
```

Comment:
create getAgent method in the service and remove one of the if statements to reduce the complexity here

Reply:
Done